runtime/gc_glue_java/ArrayletObjectModel.cpp

```diff
@@ -176,10 +182,17 @@ GC_ArrayletObjectModel::isDataAdjacentToHeader(uintptr_t dataSizeInBytes)
 }
 
 bool
-GC_ArrayletObjectModel::shouldFixupDataAddrForContiguous(J9IndexableObject *arrayPtr)
+GC_ArrayletObjectModel::shouldFixupDataAddrForContiguous(J9IndexableObject *arrayPtr, void *dataAddr)
```

Comment:
let's version this method:
- for Compactor, we use a simple variant where we pass only J9IndexableObject
- for CopyForward, we pass ForwardedHeader and dataAddr

By passing FH, we indicate that this memory should not be treated as an object (since it does not have valid class). Add that as a comment in the code.